app/controllers/visualize_controller.rb

```diff
@@ -22,6 +22,9 @@ class VisualizeController < ApplicationController
       controller.send :authenticate_actions_for_role, UserCourseMembership::ROLE_STUDENT,
                                                       course: @course,
                                                       only: [ ]
+      controller.send :authenticate_actions_for_role, UserCourseMembership::ROLE_GUEST,
```

Comment:
Possible syntax for `.send` to collapse such dual cases into a single one?  No worries if not.

Reply:
I tried but it gave errors. So, I think it is not possible to collapse such dual cases.